apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.usecase.ts

```diff
@@ -1,39 +1,149 @@
+      0,
+      workflow.steps.findIndex((stepItem) => stepItem._id === command.stepId)
+    );
+
+    return { currentStep, previousSteps };
   }
 }
 
-export const getStepDefaultOutput = (stepType: StepType): SchemaOutput => {
-  return StepTypeToOutputMap[stepType];
+const isGetByStepType = (command: GetStepSchemaCommand): command is GetStepTypeSchemaCommand =>
+  (command as GetStepTypeSchemaCommand).stepType !== undefined;
+
+const isGetByExistingStep = (command: GetStepSchemaCommand): command is GetExistingStepSchemaCommand =>
+  (command as GetExistingStepSchemaCommand).stepId !== undefined &&
+  (command as GetExistingStepSchemaCommand).workflowId !== undefined;
+
+export const buildControlsSchema = (stepType: StepType): Schema => {
+  return {
+    ...mapStepTypeToOutput[stepType],
+    description: 'Output of the step, including any controls defined in the Bridge App',
+  };
 };
 
-export const StepTypeToOutputMap = {
-  [ChannelStepEnum.SMS]: channelStepSchemas[ChannelStepEnum.SMS].output,
-  [ChannelStepEnum.EMAIL]: channelStepSchemas[ChannelStepEnum.EMAIL].output,
-  [ChannelStepEnum.PUSH]: channelStepSchemas[ChannelStepEnum.PUSH].output,
-  [ChannelStepEnum.CHAT]: channelStepSchemas[ChannelStepEnum.CHAT].output,
-  [ChannelStepEnum.IN_APP]: channelStepSchemas[ChannelStepEnum.IN_APP].output,
-  [ActionStepEnum.DELAY]: actionStepSchemas[ActionStepEnum.DELAY].output,
-  [ActionStepEnum.DIGEST]: actionStepSchemas[ActionStepEnum.DIGEST].output,
-  [ActionStepEnum.CUSTOM]: {
+const buildSubscriberSchema = () =>
+  ({
+    type: 'object',
+    description: 'Schema representing the subscriber entity',
+    properties: {
+      firstName: { type: 'string', description: "Subscriber's first name" },
+      lastName: { type: 'string', description: "Subscriber's last name" },
+      email: { type: 'string', description: "Subscriber's email address" },
+      phone: { type: 'string', description: "Subscriber's phone number (optional)" },
+      avatar: { type: 'string', description: "URL to the subscriber's avatar image (optional)" },
+      locale: { type: 'string', description: 'Locale for the subscriber (optional)' },
+      subscriberId: { type: 'string', description: 'Unique identifier for the subscriber' },
+      isOnline: { type: 'boolean', description: 'Indicates if the subscriber is online (optional)' },
+      lastOnlineAt: {
+        type: 'string',
+        format: 'date-time',
+        description: 'The last time the subscriber was online (optional)',
+      },
+    },
+    required: ['firstName', 'lastName', 'email', 'subscriberId'],
+    additionalProperties: false,
+  }) as const satisfies Schema;
+
+const buildPayloadSchema = () =>
+  ({
     type: 'object',
     properties: {},
     required: [],
     additionalProperties: false,
-  } as const satisfies Schema,
+    description:
+      'Payload Schema - For Developers. Passed during the novu.trigger method, and controlled by the developer.',
+  }) as const satisfies Schema;
```

Comment:
Didn’t we agree that this is irrelevant and not currently used? Shouldn’t we consider removing it completely?

Reply:
yeah i added it before we agreed on this, i think it wont be harmful at the moment. nevertheless i will remove it so we will add it once it is needed 👍